public/video-ui/src/util/removeStringTagDuplicates.js

```diff
@@ -0,0 +1,40 @@
+export default function removeStringTagDuplicates(tag, tagValue) {
+  // Remove string input that appears in the last words of the tag input that is being added
+  // If a word Mary has already been added and a user adds a tag with title `Mary Smith`, `Mary`
+  // gets removed from the input. Instead of the byline reading `Mary Mary Smith`, it only
+  // reads `Mary Smith`.
+
+  const tagWords = tag.webTitle
+    .split(' ')
+    .map(word => word.toLowerCase())
+    .reverse();
+
+  const addedValues = tagValue.slice(0).map(word => {
+    if (!word.id) {
+      return word.toLowerCase();
+    }
+    return word;
+  });
+
+  const valuesLength = addedValues.length;
+  let numberOfMatches = 0;
+
+  const firstTagMatch = tagWords.findIndex(tagWord => {
+    return addedValues[valuesLength - 1] === tagWord;
+  });
+
+  if (firstTagMatch !== -1) {
+    numberOfMatches = 1;
+    for (var i = firstTagMatch + 1; i < tagWords.length; i++) {
```

Comment:
I want to break out of the loop immediately when a match is not found (e.g. if someone has typed in 'Mary Smith' but picks a tag with web title 'Mary Jones', we don't want to trim any words from the input string). I guess it could be done with a foreach as well, but I think this is simpler.